app/models/SubscriptionForDACRequest.scala

```diff
@@ -27,14 +27,20 @@ import scala.util.Random
 
 case class OrganisationDetails(organisationName: String)
 object OrganisationDetails {
-  def buildOrganisationDetails(userAnswers: UserAnswers): OrganisationDetails = {
-    (userAnswers.get(ContactNamePage), userAnswers.get(SoleTraderNamePage)) match {
-      case (Some(name), _) => new OrganisationDetails(s"${name.firstName} ${name.secondName}")
-      case (_, Some(name)) => new OrganisationDetails(s"${name.firstName} ${name.secondName}")
+  def buildPrimaryContact(userAnswers: UserAnswers): OrganisationDetails = {
+    userAnswers.get(ContactNamePage) match {
+      case Some(name) => new OrganisationDetails(s"${name.firstName} ${name.secondName}")
       case _ => throw new Exception("Contact name page is empty when creating a subscription for organisation")
     }
   }
 
+  def buildSecondaryContact(userAnswers: UserAnswers): OrganisationDetails = {
+    userAnswers.get(SecondaryContactNamePage) match {
+      case Some(name) => new OrganisationDetails(name)
+      case None => throw new Exception("Secondary contact name page is empty after user answered 'Yes' to second contact")
```

Comment:
It would be better to use an Either here and throw an exception/handle it if you need to at the controller level